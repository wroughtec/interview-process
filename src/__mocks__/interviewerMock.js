export const mockData = {
  results: [
    {
      name: { first: 'firstName0', last: 'lastName0' },
      location: {
        city: 'city'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e0'
      },
      picture: {
        thumbnail: 'thumbnail'
      }
    },
    {
      name: { first: 'firstName1', last: 'lastName1' },
      location: {
        city: 'city1'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e1'
      },
      picture: {
        thumbnail: 'thumbnail1'
      }
    },
    {
      name: { first: 'firstName2', last: 'lastName2' },
      location: {
        city: 'city2'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e2'
      },
      picture: {
        thumbnail: 'thumbnail2'
      }
    },
    {
      name: { first: 'firstName3', last: 'lastName3' },
      location: {
        city: 'city3'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e3'
      },
      picture: {
        thumbnail: 'thumbnail3'
      }
    },
    {
      name: { first: 'firstName4', last: 'lastName4' },
      location: {
        city: 'city4'
      },
      login: {
        uuid: '0a985e0b-9383-43d7-bcf3-f8b8f597451e4'
      },
      picture: {
        thumbnail: 'thumbnail4'
      }
    }
  ]
};
