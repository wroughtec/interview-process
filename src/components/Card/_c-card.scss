.c-card {
  padding: 1rem;
  margin: .5rem;

  background-color: var(--cardBackground);
}

.c-card__details {
  display: flex;
  margin-bottom: 1rem;

  flex-wrap: wrap;
  align-items: center;
}

.c-card__img {
  width: 3.5rem;
  height: 3.5rem;
  padding: .25rem;
  margin-right: 1rem;

  border: 1px solid var(--imgFrame);
  border-radius: 50%;
}

.c-card__btns {
  display: flex;
}

.c-card__btn {
  display: flex;

  padding: .75rem;

  align-items: center;

  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;

  color: var(--appResetBtnText);
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;

  user-select: none;

  background-color: var(--cardBtnBackground);
  border: 1px solid transparent;
  border-color: var(--cardBtnBackground);
  border-radius: .25rem;

  transition: color .15s ease-in-out, background-color .15s ease-in-out, border-color .15s ease-in-out,
    box-shadow .15s ease-in-out;

  &:hover {
    text-decoration: none;

    background-color: var(--cardBtnBackgroundHover);
    border-color: var(--cardBtnBackgroundHover);
  }

  &:focus {
    outline: none;
    box-shadow: var(--cardBtnFocusShadow);
  }

  &:not(:disabled):not(.disabled) {
    cursor: pointer;
  }
}

.c-card__btn--right {
  margin-left: auto;
}
