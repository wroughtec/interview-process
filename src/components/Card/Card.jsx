// @flow
import React, { Component } from 'react';
import { Icon } from 'components/Icon/Icon';
import { arrowRight, arrowLeft } from 'consts/iconIds';
import './_c-card.scss';

type Props = {
  name: string,
  img: string,
  location: string,
  currentPosition: string,
  id: string,
  handleBtnClick: Function,
  btnLeft?: string,
  btnRight?: string
};

export class Card extends Component<Props> {
  handleLeftClick = () => {
    const { id, currentPosition, btnLeft, handleBtnClick } = this.props;

    return handleBtnClick(id, currentPosition, btnLeft);
  };

  handleRightClick = () => {
    const { id, currentPosition, btnRight, handleBtnClick } = this.props;

    return handleBtnClick(id, currentPosition, btnRight);
  };

  render() {
    const { name, img, location, btnLeft, btnRight } = this.props;

    return (
      <article className="c-card">
        <div className="c-card__details">
          {img && <img className="c-card__img" alt={name} title={name} src={img} />}
          <div className="c-card__body">
            {name}
            <br />
            {location}
          </div>
        </div>
        <div className="c-card__btns">
          {btnLeft && (
            <button onClick={this.handleLeftClick} className="c-card__btn c-card__btn--left" type="button">
              <Icon iconId={arrowLeft} />
            </button>
          )}
          {btnRight && (
            <button onClick={this.handleRightClick} className="c-card__btn c-card__btn--right" type="button">
              <Icon iconId={arrowRight} />
            </button>
          )}
        </div>
      </article>
    );
  }
}

Card.defaultProps = {
  btnLeft: '',
  btnRight: ''
};
