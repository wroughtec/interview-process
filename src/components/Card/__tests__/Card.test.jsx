import React from 'react';
import { shallow } from 'enzyme';
import { Card } from 'components/Card/Card';
import { mockData } from 'mocks/interviewerMock';

describe('Card', () => {
  let component, props;
  beforeEach(() => {
    props = {
      name: `${mockData.results[0].name.first} ${mockData.results[0].name.last}`,
      img: `${mockData.results[0].picture.thumbnail}`,
      location: `${mockData.results[0].location.city}`,
      currentPosition: 'currentPosition',
      id: `${mockData.results[0].login.uuid}`,
      handleBtnClick: jest.fn(),
      btnLeft: 'btnLeft',
      btnRight: 'btnRight'
    };

    component = shallow(<Card {...props} />);
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('click function should have been called', () => {
    component
      .find('button')
      .at(0)
      .simulate('click');

    expect(props.handleBtnClick).toHaveBeenCalled();

    component
      .find('button')
      .at(1)
      .simulate('click');

    expect(props.handleBtnClick).toHaveBeenCalledTimes(2);
  });
});
