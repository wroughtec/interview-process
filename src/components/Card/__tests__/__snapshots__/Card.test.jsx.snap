// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    btnLeft="btnLeft"
    btnRight="btnRight"
    currentPosition="currentPosition"
    handleBtnClick={[MockFunction]}
    id="0a985e0b-9383-43d7-bcf3-f8b8f597451e0"
    img="thumbnail"
    location="city"
    name="firstName0 lastName0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-card__details"
        >
          <img
            alt="firstName0 lastName0"
            className="c-card__img"
            src="thumbnail"
            title="firstName0 lastName0"
          />
          <div
            className="c-card__body"
          >
            firstName0 lastName0
            <br />
            city
          </div>
        </div>,
        <div
          className="c-card__btns"
        >
          <button
            className="c-card__btn c-card__btn--left"
            onClick={[Function]}
            type="button"
          >
            <Icon
              iconClass=""
              iconId="arrowLeft"
              iconSize=""
            />
          </button>
          <button
            className="c-card__btn c-card__btn--right"
            onClick={[Function]}
            type="button"
          >
            <Icon
              iconClass=""
              iconId="arrowRight"
              iconSize=""
            />
          </button>
        </div>,
      ],
      "className": "c-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="firstName0 lastName0"
              className="c-card__img"
              src="thumbnail"
              title="firstName0 lastName0"
            />,
            <div
              className="c-card__body"
            >
              firstName0 lastName0
              <br />
              city
            </div>,
          ],
          "className": "c-card__details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "firstName0 lastName0",
              "className": "c-card__img",
              "src": "thumbnail",
              "title": "firstName0 lastName0",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "firstName0 lastName0",
                <br />,
                "city",
              ],
              "className": "c-card__body",
            },
            "ref": null,
            "rendered": Array [
              "firstName0 lastName0",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "city",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="c-card__btn c-card__btn--left"
              onClick={[Function]}
              type="button"
            >
              <Icon
                iconClass=""
                iconId="arrowLeft"
                iconSize=""
              />
            </button>,
            <button
              className="c-card__btn c-card__btn--right"
              onClick={[Function]}
              type="button"
            >
              <Icon
                iconClass=""
                iconId="arrowRight"
                iconSize=""
              />
            </button>,
          ],
          "className": "c-card__btns",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                iconClass=""
                iconId="arrowLeft"
                iconSize=""
              />,
              "className": "c-card__btn c-card__btn--left",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconClass": "",
                "iconId": "arrowLeft",
                "iconSize": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                iconClass=""
                iconId="arrowRight"
                iconSize=""
              />,
              "className": "c-card__btn c-card__btn--right",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "iconClass": "",
                "iconId": "arrowRight",
                "iconSize": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-card__details"
          >
            <img
              alt="firstName0 lastName0"
              className="c-card__img"
              src="thumbnail"
              title="firstName0 lastName0"
            />
            <div
              className="c-card__body"
            >
              firstName0 lastName0
              <br />
              city
            </div>
          </div>,
          <div
            className="c-card__btns"
          >
            <button
              className="c-card__btn c-card__btn--left"
              onClick={[Function]}
              type="button"
            >
              <Icon
                iconClass=""
                iconId="arrowLeft"
                iconSize=""
              />
            </button>
            <button
              className="c-card__btn c-card__btn--right"
              onClick={[Function]}
              type="button"
            >
              <Icon
                iconClass=""
                iconId="arrowRight"
                iconSize=""
              />
            </button>
          </div>,
        ],
        "className": "c-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="firstName0 lastName0"
                className="c-card__img"
                src="thumbnail"
                title="firstName0 lastName0"
              />,
              <div
                className="c-card__body"
              >
                firstName0 lastName0
                <br />
                city
              </div>,
            ],
            "className": "c-card__details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "firstName0 lastName0",
                "className": "c-card__img",
                "src": "thumbnail",
                "title": "firstName0 lastName0",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "firstName0 lastName0",
                  <br />,
                  "city",
                ],
                "className": "c-card__body",
              },
              "ref": null,
              "rendered": Array [
                "firstName0 lastName0",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "city",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="c-card__btn c-card__btn--left"
                onClick={[Function]}
                type="button"
              >
                <Icon
                  iconClass=""
                  iconId="arrowLeft"
                  iconSize=""
                />
              </button>,
              <button
                className="c-card__btn c-card__btn--right"
                onClick={[Function]}
                type="button"
              >
                <Icon
                  iconClass=""
                  iconId="arrowRight"
                  iconSize=""
                />
              </button>,
            ],
            "className": "c-card__btns",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  iconClass=""
                  iconId="arrowLeft"
                  iconSize=""
                />,
                "className": "c-card__btn c-card__btn--left",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconClass": "",
                  "iconId": "arrowLeft",
                  "iconSize": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  iconClass=""
                  iconId="arrowRight"
                  iconSize=""
                />,
                "className": "c-card__btn c-card__btn--right",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconClass": "",
                  "iconId": "arrowRight",
                  "iconSize": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
